index.js

```diff
@@ -6,42 +6,46 @@ var hyperdrive = require('hyperdrive')
 var createSwarm = require('hyperdrive-archive-swarm')
 var raf = require('random-access-file')
 var each = require('stream-each')
-var thunky = require('thunky')
 var extend = require('xtend')
 var importFiles = require('./lib/count-import')
 var getDb = require('./lib/db')
 
-module.exports = Dat
+module.exports = function (dir, opts, cb) {
+  if (!dir) throw new Error('dir option required')
+  if (typeof opts === 'function') {
+    cb = opts
+    opts = {}
+  }
+  return new Dat(dir, opts, cb)
+}
 
-function Dat (opts) {
-  if (!(this instanceof Dat)) return new Dat(opts)
+function Dat (dir, opts, cb) {
```

Comment:
`dir` was in the options because I wanted to (eventually) make it optional, not required, for the downloads (since we talked about downloading to the directory name. This makes more sense for how it works now but we'd have to break this down the line if we want to do the optional download dir.